hermes-common/src/main/java/pl/allegro/tech/hermes/common/hook/HooksHandler.java

```diff
@@ -21,24 +23,35 @@ public void addShutdownHook(ServiceAwareHook hook) {
     }
 
     public void shutdown(ServiceLocator serviceLocator) {
+        deregisterGlobalShutdownHook();
```

Comment:
It is in this package/class because global shutdown hook is registered here. It is pragmatic fix. If would you like do it in different way, refactor something please let me know.